music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
+        ...                           metadata.Contributor(role='original partial plot by',
+        ...                               name='Friedrich Schiller'))
+        >>> md.addCustom('composer', 'Not a contributor')
+        >>> allContributors = md.getAllContributorNamedValues()
+        >>> allContributors
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Antonín Dvořák>),
+        ('marcrel:LBT',
+        <music21.metadata.primitives.Contributor librettist:Marie Červinková-Riegrová>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor based on plot by:Ferdinand Mikovec>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor original partial plot by:Friedrich Schiller>)]
+        '''
+
+        allOut: t.List[t.Tuple[str, ValueType]] = []
+
+        for nsKey, value in self._metadata.items():
+            if not self._isContributorNSKey(nsKey):
+                continue
+
+            for v in value:
+                allOut.append((nsKey, v))
+
+        return allOut
+
+# -----------------------------------------------------------------------------
+#   A few static utility routines for clients calling public APIs
+
+    @staticmethod
+    def uniqueNameToNSKey(uniqueName: str) -> t.Optional[str]:
+        '''
+        Translates a unique name to the associated standard property's NSKey.
+        Returns None if no such associated standard property can be found.
+
+        >>> metadata.Metadata.uniqueNameToNSKey('librettist')
+        'marcrel:LBT'
+        >>> metadata.Metadata.uniqueNameToNSKey('not a standard property') is None
+        True
+        >>> metadata.Metadata.uniqueNameToNSKey('alternativeTitle')
+        'dcterms:alternative'
+        '''
+        if not isinstance(uniqueName, str):
+            raise ValueError('uniqueName must be str')
+
+        return properties.UNIQUE_NAME_TO_NSKEY.get(uniqueName, None)
+
+    @staticmethod
+    def isContributorUniqueName(uniqueName: str) -> bool:
+        '''
+        Determines if a unique name is associated with a standard contributor
+        property.  Returns False if no such associated standard property can
+        be found.
+
+        >>> metadata.Metadata.isContributorUniqueName('librettist')
+        True
+        >>> metadata.Metadata.isContributorUniqueName('otherContributor')
+        True
+        >>> metadata.Metadata.isContributorUniqueName('alternativeTitle')
+        False
+        >>> metadata.Metadata.isContributorUniqueName('not a standard property')
+        False
+        >>> metadata.Metadata.isContributorUniqueName(None)
+        False
+        '''
+        if not uniqueName:
+            return False
+        prop: PropertyDescription = (
+            properties.UNIQUE_NAME_TO_PROPERTY_DESCRIPTION.get(uniqueName, None))
+        if prop is None:
+            return False
+
+        return prop.isContributor
+
+    @staticmethod
+    def _isStandardUniqueName(uniqueName: str) -> bool:
```

Comment:
I don't understand the logic for why the prior staticmethod is public but these are private.  It seems like all or nothing.

Reply:
Yes, I need to make final pass at public vs private.  I was only making public what other parts of music21 were using to begin with.

Reply:
Done.